libcaf_core/caf/detail/append_uri.hpp

```diff
@@ -0,0 +1,31 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include "caf/detail/append_hex.hpp"
+
+namespace caf {
+namespace detail {
+
+// Escapes all reserved characters according to RFC 3986 in `x` and
+// adds the encoded string to `str`.
+void append_uri(std::string& str, string_view x, bool is_path = false);
```

Comment:
`append_uri` does not make a lot of sense to me, since it takes a `string_view` instead of an URI. Either we should make the context clear via an extra suffix or (my vote) we use the actual name of encoding: [`append_percent_encoded`](https://en.wikipedia.org/wiki/Percent-encoding).